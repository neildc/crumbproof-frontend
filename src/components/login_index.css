
.loginCard {
    margin: 0 auto;
    text-align: center;
}

@media (min-width:768px) {
    .loginCard {
        width: 50%;
        height: 50%;
    }

    .loginForm {
        padding:50px;
    }
}

.loginCardTitle {
    text-align: left;
    backgroundColor: #EEE;
}

.loginButton {
    margin-bottom: 20px;
    margin-top: 20px;
}
