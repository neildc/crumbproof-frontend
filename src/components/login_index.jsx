
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Field, reduxForm } from 'redux-form';

import RaisedButton from 'material-ui/RaisedButton';
import Snackbar from 'material-ui/Snackbar';
import TextField from 'material-ui/TextField';
import { Card, CardTitle } from 'material-ui/Card';

import { authLogin, authClearError } from '../actions/actions_auth';

import SubmitButton from './submit_button';
import { FadeIn } from './animations/fade';
import { required } from '../validators';

import './login_index.css';

class LoginIndex extends Component {
  renderField(field) {
    const { meta: { touched, error } } = field;

    return (
      <div>
        <TextField
          style={{ width: 'auto' }}
          hintText=""
          floatingLabelText={field.label}
          errorText={touched && error}
          {...field.input}
          {...field.custom}
          type={field.type}
        />
      </div>
    );
  }

  onSubmit(values) {
    return this.props.authLogin(values, () => {
      this.props.history.push('/');
    });
  }

  render() {
    const { handleSubmit, submitting } = this.props;

    return (
      <FadeIn>
        <Card className="loginCard">
          <CardTitle title="Login" className="loginCardTitle" />

          <form
            onSubmit={handleSubmit(this.onSubmit.bind(this))}
            className="loginForm"
          >

            <Field
              label="Username"
              name="username"
              component={this.renderField}
              validate={[required]}
            />
            <Field
              label="Password"
              name="password"
              type="password"
              component={this.renderField}
              validate={[required]}
            />

            <SubmitButton
              className="loginButton"
              label="Login"
              labelInProgress="Logging in..."
              submittingFlag={submitting}
              fullWidth
            />
            <RaisedButton
              type="button"
              label="Register"
              fullWidth
              containerElement={<Link to="/register" />}
            />

          </form>

          <Snackbar
            open={this.props.error != null}
            message={this.props.error ? this.props.error : ''}
            autoHideDuration={5000}
            style={{ backgroundColor: 'red' }}
            onRequestClose={this.props.authClearError}
          />
        </Card>
      </FadeIn>
    );
  }
}

function mapStateToProps({ auth }, ownProps) {
  return { error: auth.error };
}

export default reduxForm({
  form: 'LoginForm',
})(connect(mapStateToProps, { authLogin, authClearError })(LoginIndex));
