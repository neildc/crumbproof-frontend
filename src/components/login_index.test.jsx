import React from 'react';
import { shallow } from 'enzyme';
import { LoginIndex } from './login_index';

it('renders without throwing', () => {
  const component = shallow(
    <LoginIndex
      error={null}
      handleSubmit={jest.fn()}
      submitting={false}
    />,
  );

  expect(component).toMatchSnapshot();
});
