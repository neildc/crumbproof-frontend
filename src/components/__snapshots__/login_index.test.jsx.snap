// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without throwing 1`] = `
<FadeIn>
  <Card
    className="loginCard"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardTitle
      className="loginCardTitle"
      title="Login"
    />
    <form
      className="loginForm"
    >
      <Field
        component={[Function]}
        label="Username"
        name="username"
        validate={
          Array [
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        label="Password"
        name="password"
        type="password"
        validate={
          Array [
            [Function],
          ]
        }
      />
      <SubmitButton
        className="loginButton"
        fullWidth={true}
        label="Login"
        labelInProgress="Logging in..."
        submittingFlag={false}
      />
      <RaisedButton
        containerElement={
          <Link
            replace={false}
            to="/register"
          />
        }
        disabled={false}
        fullWidth={true}
        label="Register"
        labelPosition="after"
        primary={false}
        secondary={false}
        type="button"
      />
    </form>
    <Snackbar
      autoHideDuration={5000}
      message=""
      open={false}
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    />
  </Card>
</FadeIn>
`;
